puma/tests/test_yuma.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+"""Unit test script for the functions in hlplots/tagger.py."""
+from __future__ import annotations
+
+import tempfile
+import unittest
+from dataclasses import dataclass
+from pathlib import Path
+
+import h5py
+import numpy as np
+import yaml
+from ftag import Flavours, get_mock_file
+
+from puma.plot_ftag import main
+from puma.yuma import PlotConfig, get_signals
+from puma.yuma.yutils import get_tagger_name
+
+
+def structured_from_dict(d: dict[str, np.ndarray]) -> np.ndarray:
```

Comment:
this function is already in the tools package: https://github.com/umami-hep/atlas-ftag-tools/blob/0a0d5948ff324e97c405443026fcd0ecbec77bb8/ftag/hdf5/h5utils.py#L112